app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,43 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.delete unless queue_item.user != current_user
+    update_queue_items_positions()
+    redirect_to my_queue_path
+  end
+
+  private
+
+  def queue_video(video)
+    QueueItem.create(video: video, user: current_user, position: new_queue_item_position) unless current_user_already_queued_this_video?(video)
+  end
+
+  def new_queue_item_position
+    current_user.queue_items.count + 1
+  end
+
+  def current_user_already_queued_this_video?(video)
+    current_user.queue_items.map{|queue_item| queue_item.video}.include?(video)
+  end
+
+  def update_queue_items_positions
+    queue_items = current_user.queue_items
+    queue_items = queue_items.sort_by{|queue_item| queue_item.position}
```

Comment:
I would recommend sorting queue_items at source where you define `has_many :queue_items`, the reason for this would be you want to have it sorted based on position on other different places and it would produce less code.